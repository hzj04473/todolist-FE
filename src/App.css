body {
  color: lightblue !important ;
  background-color: aliceblue !important;
}

.todo-item {
  border: 1px solid lightblue;
  background-color: white;
  display: flex;
  justify-content: space-between;
  align-items: center;
}

.todo-content {
  padding: 1em;
}

.add-item-row {
  margin-top: 50px !important;
  margin-bottom: 30px;
}
.input-box {
  padding: 10px;
  width: 100%;
  border: 1px solid lightblue;
}
.input-box:focus {
  outline: none;
}
.button-add {
  height: 85%;
  width: 100%;
  border: none;
  background-color: lightblue;
  color: white;
  transition: background-color 0.3s;
  /* padding: 1rem 0.75rem; */
}
.button-add:hover {
  background-color: white;
  color: lightblue;
  border: 2px solid lightblue;
}
.button-add.btn-active {
  background-color: lightblue !important;
  color: white !important;
}

.button-delete {
  background-color: white;
  border: 2px solid lightblue;
  color: lightblue;
  margin: 10px;
}
.button-delete:hover {
  background-color: lightblue;
  color: #ffffff;
}

.item-complete {
  background-color: gainsboro;
  color: white;
}

.item-complete.cancel-line {
  text-decoration: line-through;
  text-decoration-color: red;
  text-decoration-style: wavy;
}

.login-box {
  border: 1px solid lightblue;
  max-width: 600px;
  width: 100%;
  background-color: white;
  padding: 20px;
}
.login-box label {
  font-weight: 700;
}

.display-center {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  height: 100vh;
}

.button-box {
  display: flex;
  justify-content: space-between;
}

/* .button-primary {
  background-color: lightblue;
  color: #ffffff;
  border: none;
} */

.red-error {
  color: red;
}
.text-truncate-multiline {
  min-height: 48px; /* 최소 높이 설정 */
  max-height: 110px; /* 최대 높이 설정 */
  overflow: hidden;
  /* display: -webkit-box; */
  display: block;
  line-clamp: 2; /* 표준 속성 */
  -webkit-line-clamp: 2;
  box-orient: vertical; /* 표준 속성 */
  text-overflow: ellipsis;
  word-wrap: break-word;
}

.text-truncate-title-multiline {
  min-height: 50px; /* 최소 높이 설정 */
  max-height: 70px; /* 최대 높이 설정 */
  overflow: hidden;
  /* display: -webkit-box; */
  display: block;
  line-clamp: 2; /* 표준 속성 */
  -webkit-line-clamp: 2;
  box-orient: vertical; /* 표준 속성 */
  text-overflow: ellipsis;
  word-wrap: break-word;
}

.custom-button:disabled {
  background-color: #d6d6d6; /* 비활성화된 배경색 */
  color: #888; /* 비활성화된 텍스트 색 */
  border-color: #ccc; /* 비활성화된 테두리 색 */
  cursor: not-allowed; /* 마우스 커서 */
}
