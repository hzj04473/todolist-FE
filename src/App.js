import 'bootstrap/dist/css/bootstrap.min.css';
import './App.css';
import { Route, Routes } from 'react-router-dom';
import TodoPage from './pages/TodoPage';
import RegisterPage from './pages/RegisterPage';
import LoginPage from './pages/LoginPage';
import { useCallback, useEffect, useState } from 'react';
import PrivateRoute from './route/PrivateRoute';
import api from './utils/api';
import { Container } from 'react-bootstrap'; // 추가
import { HelmetProvider } from 'react-helmet-async'; // ✅ HelmetProvider 추가

function App() {
  const [user, setUser] = useState(null);
  const getUser = useCallback(async () => {
    try {
      const storedToken = sessionStorage.getItem('token');
      if (storedToken) {
        const response = await api.get('/user/me');
        setUser(response.data.user);
      }
    } catch (error) {
      setUser(null);
    }
  }, []);

  useEffect(() => {
    getUser();
  }, [getUser]);

  return (
    <HelmetProvider>
      <Container fluid className="p-0">
        {/* 추가: 전체 레이아웃을 Container로 감싸기 */}
        <Routes>
          {/* Private Router */}
          {/* 메인페이지 */}
          <Route
            path="/"
            element={
              <PrivateRoute user={user}>
                <TodoPage user={user} setUser={setUser} />
              </PrivateRoute>
            }
          />
          {/* 검색할떄 */}
          <Route
            path="/search/:keyword"
            element={
              <PrivateRoute user={user}>
                <TodoPage user={user} setUser={setUser} />
              </PrivateRoute>
            }
          />
          {/* 회원가입 / 수정 */}
          <Route
            path="/register"
            element={<RegisterPage user={user} setUser={setUser} />}
          />

          {/* 로그인페이지 */}
          <Route
            path="/login"
            element={<LoginPage user={user} setUser={setUser} />}
          />

          {/* 카카오톡 Redirect_url */}
          <Route
            path="/auth/kakao/callback"
            element={<RegisterPage snsLoginType={`kakao`} />}
          />

          <Route
            path="*"
            element={
              <PrivateRoute user={user}>
                <TodoPage user={user} setUser={setUser} />
              </PrivateRoute>
            }
          />
        </Routes>
      </Container>
    </HelmetProvider>
  );
}

export default App;
