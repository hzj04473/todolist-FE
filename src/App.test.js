import { render } from '@testing-library/react';
import { HelmetProvider } from 'react-helmet-async';
import App from './App';

// API 모킹
jest.mock('./utils/api', () => ({
  get: jest.fn(() =>
    Promise.resolve({
      status: 200,
      data: { data: [] },
    })
  ),
  post: jest.fn(),
  put: jest.fn(),
  delete: jest.fn(),
}));

describe('App Component', () => {
  it('matches snapshot', async () => {
    const { container } = render(
      <HelmetProvider>
        <App />
      </HelmetProvider>
    );

    expect(container).toMatchSnapshot();
  });
});
