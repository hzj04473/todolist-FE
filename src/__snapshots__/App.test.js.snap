// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component matches snapshot 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="add-item-row row"
    >
      <div
        class="col-sm-10 col-12"
      >
        <input
          class="input-box"
          placeholder="할일을 입력하세요"
          type="text"
          value=""
        />
      </div>
      <div
        class="col-sm-2 col-12"
      >
        <button
          class="button-add"
        >
          추가
        </button>
      </div>
    </div>
    <div>
      <h2>
        Todo List
      </h2>
      <div>
        There is no Item to show
      </div>
    </div>
  </div>
</div>
`;
